File: ts/project/.sdk/src/cmp/swift/fragment/EntityUpdateOp.fragment.swift
// Update operation fragment (spliced into the entity class by
// EntityOperation_swift; only the EJECT region is emitted).

import Foundation

final class EntyClassUpdateOpFragment: EntyClass {
// EJECT-START
public override func update(_ reqdata: VMap?, _ ctrl: VMap?) throws -> Value {
  var ctxmap: [String: Any?] = ["opname": "update", "match": match, "data": data]
  if let ctrl = ctrl { ctxmap["ctrl"] = ctrl }
  if let reqdata = reqdata { ctxmap["reqdata"] = reqdata }
  let ctx = utility.makeContext(ctxmap, entctx)

  return try runOp(ctx) {
    if let result = ctx.result {
      if let rm = result.resmatch { self.match = rm }
      if !isNil(result.resdata) {
        self.data = clone(result.resdata).asMap ?? VMap()
      }
    }
  }
}
// EJECT-END
}

File: ts/project/.sdk/src/cmp/csharp/fragment/EntityUpdateOp.fragment.cs
// Update operation fragment (spliced into the entity class by
// EntityOperation_csharp; only the EJECT region is emitted).

namespace ProjectNameSdk.Entity;

public class EntyClassUpdateOpFragment : EntyClass
{
    public EntyClassUpdateOpFragment(ProjectNameSDK client) : base(client) { }

// EJECT-START
public override object? Update(Dictionary<string, object?>? reqdata,
    Dictionary<string, object?>? ctrl = null)
{
    var ctx = utility.MakeContext(new Dictionary<string, object?>
    {
        ["opname"] = "update",
        ["ctrl"] = ctrl,
        ["match"] = match,
        ["data"] = data,
        ["reqdata"] = reqdata,
    }, entctx);

    return RunOp(ctx, () =>
    {
        if (ctx.Result != null)
        {
            if (ctx.Result.Resmatch != null)
            {
                match = ctx.Result.Resmatch;
            }
            if (ctx.Result.Resdata != null)
            {
                data = Helpers.ToMapAny(
                    Voxgig.Struct.StructUtils.Clone(ctx.Result.Resdata))
                    ?? new Dictionary<string, object?>();
            }
        }
    });
}
// EJECT-END
}
